packages/grid/_modules_/grid/hooks/utils/useLogger.ts

```diff
@@ -46,7 +46,13 @@ function getAppender(name: string, logLevel: string, appender: Logger = console)
     if (idx >= minLogLevelIdx) {
       loggerObj[method] = (...args: any[]) => {
         const [message, ...rest] = args;
-        (appender as any)[method](`[${name}] - ${message}`, ...rest);
+
+        // Don't console log non errors in production
+        if (appender === console && method !== 'error' && process.env.NODE_ENV === 'production') {
```

Comment:
that's not necessary, we use the factory to initialize the logger and we want to log errors in prod as well

Reply:
I think that it's important for 1. of https://github.com/mui-org/material-ui-x/pull/350#discussion_r496013426.
If the message should be logged in production, then it should be an error, not a warning. I forgot to add `forceDebug`

```suggestion
        if (
          appender === console &&
          !forceDebug &&
          method !== 'error' &&
          process.env.NODE_ENV === 'production'
        ) {
```

Reply:
IMHO the logging level should not change between prod and dev. Otherwise ppl will have an error in prod and not in dev and they will be like WTF

Reply:
> Otherwise ppl will have an error in prod and not in dev and they will be like WTF

This can't happen in the current configuration. All the errors reported in production as also reported in development.
The opposite (warn in dev, no warn in pro) is how React and Material-UI behaves (and have been since the beginning), developers are used to it. I think that the surprise will be if we report warning in prod.

> IMHO the logging level should not change between prod and dev.

Oh, I didn't think about that. It could be cleaner. We could have the log level as warning in dev and error in production, instead of the custom forking logic. This sounds better and more consistent! I can do that.